### ISSUE TITLE
Watermark Image - V2.0.245

### ISSUE THREAD

**Issue by emanalzoaubi** [dansiegel/Mobile.BuildTools#285]
# Description

I'm trying to add a watermark on the application icon. I added a beta-version.png image as described in the documentation. However, I got the error "Invalid file name: It must contain only [^a-zA-Z0-9_.]+. C:\Users\User\source\repos\App1\App1\App1.Android\obj\Debug\110\Resources\drawable-hdpi\beta-version.png".
After that, I tried to downgrade Mobile.BuildTools library to version 1.4.0.638 and the error is gone.
Also, I changed the watermark image to be betaversion, the error of Invalid name is gone but I got a new error "Unhandled error while executing ImageResizerTask"

## Environment

- OS: [Windows 10]
- IDE: [VS2019]
- BuildTools Version: [2.0.245]
- IDE Version: [4.8.04084]
- Build Target Framework: [netstandard2.0|Xamarin.iOS|Xamarin.Android]

## Reproduction App
1. Install Mobile.Buildtools V2.0.245
2. Create the Images folder and Debug subfolder in the solution directory.
3. Edit buildtools.json file, adding 
{
  "$schema": "https://mobilebuildtools.com/schemas/v2/buildtools.schema.json",
   "images": {
    "directories": ["Images"],
    "conditionalDirectories": {
      "Debug": ["Images\\Debug"]
	  }
  }
}
4. add beta-version.png into Debug folder and icon.png to Images folder.
5. build the app
6. edit beta-version.json to be 
{
  "$schema": "http://mobilebuildtools.com/schemas/v2/resourceDefinition.schema.json",
  "ignore": true
}
7. edit icon.json to be 
{
  "$schema": "http://mobilebuildtools.com/schemas/v2/resourceDefinition.schema.json",
  "watermark": {
    "sourceFile": "beta-version"
  }
}
8. copy icon.json to Debug folder.
9. build the application and get the error Unhandled error while executing ImageResizerTask

**Comment by bijington** [comment]
@emanalzoaubi At an initial glance from your error it looks like you didn't set the beta-version.png image to be ignored by the Build Tools. It should be documented here: https://mobilebuildtools.com/images/configuring-images/

Basically Android refuses to accept - characters in image names which is why you see the error being reported. On top of that you don't actually want this watermark image to be included in your app as a separate resource which is why you need to mark it as ignored in its JSON file

**Comment by emanalzoaubi** [comment]
@bijington, Ok great!
I've followed the documentation but I've got the below error "Unhandled error while executing ImageResizerTask".
You can find "icon.json, beta-version.json & buildtools.json" in the attached.
[WatermarkTest.zip](https://github.com/dansiegel/Mobile.BuildTools/files/6991505/WatermarkTest.zip)

**Comment by bijington** [comment]
@emanalzoaubi I think there may be a mistake in the documentation. I believe you need to include the file extension when specifying a `sourceFile`. Could you try changing the following parts from your json files:

## Images/icon.json

change:

```json
{
  "$schema": "http://mobilebuildtools.com/schemas/v2/resourceDefinition.schema.json",
  "watermark": {
    "sourceFile": "beta-version"
  }
}
```

to:

```json
{
  "$schema": "http://mobilebuildtools.com/schemas/v2/resourceDefinition.schema.json",
  "watermark": {
    "sourceFile": "beta-version.png"
  }
}
```

You will also need to do the same for the `Images/Debug/icon.json` file.

**Comment by emanalzoaubi** [comment]
@bijington unfortunately, the same error occurs😫
Are there any constraints about the images size or their padding?
I tried to add the used images in the sample but with no chance 😓

**Comment by bijington** [comment]
I can confirm that I can see the same issue using your files. I have attached a full solution that shows the problem.

I will aim to take a look at why this is failing at some point

[MBT.zip](https://github.com/dansiegel/Mobile.BuildTools/files/7000309/MBT.zip)

**Comment by bijington** [comment]
This was fixed by PR #292. Do we leave this open until release or can we close it?
